File: include/correlations/wet-gas-volume-factor.rb
# coding: utf-8

begin_correlation("Bwg", "WetGasVolumeFactor", "RB_STB")
add_title("CALCULATION OF WET GAS FORMATION VOLUME FACTOR")
add_parameter("t", "Rankine", "Temperature")
add_parameter("p", "psia", "Pressure")
add_parameter("z", "Zfactor", "Gas compressibility factor")
add_parameter("rsp1", "SCF_STB", "Primary separator producing GOR")
add_parameter("veq", "SCF_STB", "Equivalent volume")
add_author("Standard Equation")
add_note("The formation volume factor of a wet gas is defined as the volume of reservoir gas required to produce one stock-tank barrel of liquid at the surface.")
add_ref("mcCain:1990")
add_internal_note("The equation was verified by using as reference: McCain (1990). Date: March 06 2017.")
end_correlation()

File: include/correlations/wet-gas-volume-factor.cc
Quantity<RCFGas_STB> Bwg__correlation__fct(const Quantity<Rankine>& t,
                                           const Quantity<psia>& p,
                                           const Quantity<Zfactor>& z,
                                           const Quantity<SCF_STB>& rsp1,
                                           const Quantity<SCF_STB>& veq) {
  return Bwg::get_instance()(t, p, z, rsp1, veq);
}

double Bwg__correlation__valid__fct(const double& t, const double& p,
                                    const double& z, const double& rsp1,
                                    const double& veq) {
  return Bwg::get_instance()(t, p, z, rsp1, veq).raw();
}
